supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala

```diff
@@ -79,83 +85,177 @@ final case class AwsBatchJob(jobDescriptor: BackendJobDescriptor, // WDL/CWL
-    val uniquePath = workflow.callable.name + "/" +
-                     jobDescriptor.taskCall.localName + "/" +
-                     workflow.id + "/" +
-                     jobDescriptor.key.index.fromIndex + "/" +
-                     jobDescriptor.key.attempt
-    Log.info(s"""Submitting job to AWS Batch""")
-    Log.info(s"""dockerImage: ${runtimeAttributes.dockerImage}""")
-    Log.info(s"""jobQueueArn: ${runtimeAttributes.queueArn}""")
-    Log.info(s"""taskId: $taskId""")
-    Log.info(s"""hostpath root: $uniquePath""")
 
+    //find or create the script in s3 to execute
+    val scriptKey = findOrCreateS3Script(reconfiguredScript, runtimeAttributes.scriptS3BucketName)
+
+    //calls the client to submit the job
     def callClient(definitionArn: String, awsBatchAttributes: AwsBatchAttributes): Aws[F, SubmitJobResponse] = {
-      val submit =
-        async.delay(client.submitJob(
+
+      Log.info(s"Submitting taskId: $taskId, job definition : $definitionArn, script: s3://${runtimeAttributes.scriptS3BucketName}/$scriptKeyPrefix$scriptKey")
+
+      val outputinfo = outputs.map(o => "%s,%s,%s,%s".format(o.name, o.s3key, o.local, o.mount))
+        .mkString(";")
+      val inputinfo = inputs.collect{case i: AwsBatchFileInput => i}
+        .map(i => "%s,%s,%s,%s".format(i.name, i.s3key, i.local, i.mount))
+        .mkString(";")
+
+      val submit: F[SubmitJobResponse] =
+        async.delay(batchClient.submitJob(
           SubmitJobRequest.builder()
             .jobName(sanitize(jobDescriptor.taskCall.fullyQualifiedName))
             .parameters(parameters.collect({ case i: AwsBatchInput => i.toStringString }).toMap.asJava)
+
+            //provide job environment variables, vcpu and memory
+            .containerOverrides(
+              ContainerOverrides.builder
+                .environment(
+                  buildKVPair("BATCH_FILE_TYPE", "script"),
+                  buildKVPair("BATCH_FILE_S3_URL",
+                    s"s3://${runtimeAttributes.scriptS3BucketName}/$scriptKeyPrefix$scriptKey"),
+                  buildKVPair("AWS_CROMWELL_CALL_ROOT", jobPaths.callExecutionRoot.toString),
+                  buildKVPair("AWS_CROMWELL_WORKFLOW_ROOT", jobPaths.workflowPaths.workflowRoot.toString),
+                  gzipKeyValuePair("AWS_CROMWELL_INPUTS", inputinfo),
+                  buildKVPair("AWS_CROMWELL_OUTPUTS", outputinfo),
+                  buildKVPair("AWS_CROMWELL_STDOUT_FILE", dockerStdout),
+                  buildKVPair("AWS_CROMWELL_STDERR_FILE", dockerStderr),
+                  buildKVPair("AWS_CROMWELL_RC_FILE", dockerRc),
+                )
+                .memory(runtimeAttributes.memory.to(MemoryUnit.MB).amount.toInt)
+                .vcpus(runtimeAttributes.cpu.##).build
+            )
             .jobQueue(runtimeAttributes.queueArn)
-            .jobDefinition(definitionArn).build
+            .jobDefinition(definitionArn)
+            // todo add JobRetryStrategy with retry attempts
```

Comment:
Is this TODO adding more functionality, or recovering functionality that used to exist in AWS but now does not?

If it's the former that's fine to leave for later. but If it's the latter, it'd be nice to cover it before merging.

Reply:
No, something that we didn't have that would be good to have in the future.